### PR TITLE
2.10.5 Release Notes

### PR DESCRIPTION
**Pull request by billpalombi**
Draft of 2.10.5 Release Notes

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+### Cloud Provider Workers
+
+Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are shipping types workers for each major cloud provider, AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively. 
+
+### Enhancements
+- In order to allow more space for the flow run graph, the metadata in right side bar of the flow run page has moved to the new "Details" tab - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
+
+### Fixes
+- fix(deployments): remove erroneous creation of work queue by @joelluijmes in https://github.com/PrefectHQ/prefect/pull/9046
+- Add `configuration` to `kill_infrastructure` signature by @desertaxle in https://github.com/PrefectHQ/prefect/pull/9250
```
Comment (serinamarie) [inline RELEASE-NOTES.md:18]:
```suggestion
- Add `configuration` to `kill_infrastructure` signature - [#9250] (https://github.com/PrefectHQ/prefect/pull/9250)
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+
+### Experimental
+- Add worker.executed-flow-run event by @bunchesofdonald in https://github.com/PrefectHQ/prefect/pull/9227
+- Emit worker lifecycle events. by @bunchesofdonald in https://github.com/PrefectHQ/prefect/pull/9249
+- Emit cancelled-flow-run event when worker cancels a flow run by @bunchesofdonald in https://github.com/PrefectHQ/prefect/pull/9255
+
+### Documentation
+- Fix broken link on docs landing page - [#9247](https://github.com/PrefectHQ/prefect/pull/9247)
+- Remove outdated warning from task run concurrency UI docs - [#9256](https://github.com/PrefectHQ/prefect/pull/9256)
+- Add button to edit documentation to each page - [#9259](https://github.com/PrefectHQ/prefect/pull/9259)
```
Comment (serinamarie) [inline RELEASE-NOTES.md:28]:
```suggestion
- Enhance ability to edit documentation - [#9259](https://github.com/PrefectHQ/prefect/pull/9259)
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+
+### Enhancements
+- In order to allow more space for the flow run graph, the metadata in right side bar of the flow run page has moved to the new "Details" tab - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
+
+### Fixes
+- fix(deployments): remove erroneous creation of work queue by @joelluijmes in https://github.com/PrefectHQ/prefect/pull/9046
+- Add `configuration` to `kill_infrastructure` signature by @desertaxle in https://github.com/PrefectHQ/prefect/pull/9250
+
+### Experimental
+- Add worker.executed-flow-run event by @bunchesofdonald in https://github.com/PrefectHQ/prefect/pull/9227
```
Comment (serinamarie) [inline RELEASE-NOTES.md:21]:
```suggestion
- Add worker.executed-flow-run event - [#9227] (https://github.com/PrefectHQ/prefect/pull/9227)
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,32 @@
...
+- Emit worker lifecycle events. by @bunchesofdonald in https://github.com/PrefectHQ/prefect/pull/9249
```
Comment (serinamarie) [inline RELEASE-NOTES.md:22]:
```suggestion
- Emit worker lifecycle events - [#9249] https://github.com/PrefectHQ/prefect/pull/9249
```

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,32 @@
...
+- Emit cancelled-flow-run event when worker cancels a flow run by @bunchesofdonald in https://github.com/PrefectHQ/prefect/pull/9255
```
Comment (serinamarie) [inline RELEASE-NOTES.md:23]:
```suggestion
- Emit `cancelled-flow-run` event when worker cancels a flow run  - [#9255] (https://github.com/PrefectHQ/prefect/pull/9255)
```

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,32 @@
...
 
+## Release 2.10.5
+
+### Deploy a Prefect flow via Github Actions
+
+With the new `Deploy a Prefect flow` Github action, you can deploy a flow to Prefect Cloud via the new [Projects](https://docs.prefect.io/latest/concepts/projects/) system.
+
+### Cloud Provider Workers
+
+Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are shipping types workers for each major cloud provider, AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively.
```
[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:11]:
```suggestion
Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are releasing workers for each major cloud provider: AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively. 
```

[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:11]: "shipping" feels weird as a user-facing verb. All workers are typed so... no reason to mention that imo.

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,32 @@
...
+
+### Cloud Provider Workers
+
+Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are shipping types workers for each major cloud provider, AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively. 
+
+### Enhancements
+- In order to allow more space for the flow run graph, the metadata in right side bar of the flow run page has moved to the new "Details" tab - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
+
+### Fixes
+- fix(deployments): remove erroneous creation of work queue by @joelluijmes in https://github.com/PrefectHQ/prefect/pull/9046
```
[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:17]: "when creating a deployment" ? When does this occur?

File: RELEASE-NOTES.md (hunk 4 above)
Comment (madkinsz) [reply RELEASE-NOTES.md:22]: Probably "Emit events for worker lifecyle"?

File: RELEASE-NOTES.md (hunk 1 above)
Comment (madkinsz) [reply RELEASE-NOTES.md:18]: Probably want `Worker.kill_infrastructure`

File: RELEASE-NOTES.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (serinamarie) [reply RELEASE-NOTES.md:17]: when creating a deployment that doesn't use the `default-agent-pool` work pool, it seems

[the remark this answers is not in the document]
Comment (serinamarie) [reply RELEASE-NOTES.md:17]: where queue was duplicated in the default-agent-pool when creating a deployment

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,5 +1,36 @@
...
+
+### Enhancements
+- Add `idempotency_key` to flow runs filter — [#8600](https://github.com/PrefectHQ/prefect/pull/8600)
+- Add details tab to flow run page and increase flow run graph width - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
+
+### Fixes
+- Fix issue in which work queues were duplicated in the `default-agent-pool` when creating a deployment - [#9046](https://github.com/PrefectHQ/prefect/pull/9046)
+- Add `configuration` to `Worker.kill_infrastructure` signature - [#9250](https://github.com/PrefectHQ/prefect/pull/9250)
+- Update `critical_service_loop` to throw a runtime error on failure — https://github.com/PrefectHQ/prefect/pull/9267
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:21]:
```suggestion
- Fix pip requirement inference compatibility with Python 3.11+ and pip 23.1+ - [#9278] (https://github.com/PrefectHQ/prefect/pull/9278)
```

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,39 @@
...
+- Emit events for worker lifecycle - [#9249](https://github.com/PrefectHQ/prefect/pull/9249)
+- Emit `cancelled-flow-run` event when worker cancels a flow run - [#9255](https://github.com/PrefectHQ/prefect/pull/9255)
+
+### Documentation
+- Fix broken link on docs landing page - [#9247](https://github.com/PrefectHQ/prefect/pull/9247)
+- Remove outdated warning from task run concurrency UI docs - [#9256](https://github.com/PrefectHQ/prefect/pull/9256)
+- Add `edit` button to docs to improve ability to fix documentation - [#9259](https://github.com/PrefectHQ/prefect/pull/9259)
+- Remove UI documentation pages, reorganize content, and simplify side bar navigation structure - [#9039](https://github.com/PrefectHQ/prefect/pull/9039)
+- Add tutorial for creating a worker - [#9179](https://github.com/PrefectHQ/prefect/pull/9179)
+- Add GitHub action to trigger versioned builds in docs repository — [#8984](https://github.com/PrefectHQ/prefect/pull/8984)
```
Comment (serinamarie) [inline RELEASE-NOTES.md:35]:
```suggestion
- Add GitHub Action to trigger versioned builds in docs repository — [#8984](https://github.com/PrefectHQ/prefect/pull/8984)
```

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,39 @@
...
+
+### Enhancements
+- Add `idempotency_key` to flow runs filter — [#8600](https://github.com/PrefectHQ/prefect/pull/8600)
+- Add details tab to flow run page and increase flow run graph width - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
+- Add status code to base client log on retry - [#9265] (https://github.com/PrefectHQ/prefect/pull/9265)
+
+### Fixes
+- Fix issue in which work queues were duplicated in the `default-agent-pool` when creating a deployment - [#9046](https://github.com/PrefectHQ/prefect/pull/9046)
+- Add `configuration` to `Worker.kill_infrastructure` signature - [#9250](https://github.com/PrefectHQ/prefect/pull/9250)
+- Update `critical_service_loop` to throw a runtime error on failure — https://github.com/PrefectHQ/prefect/pull/9267
```
Comment (serinamarie) [inline RELEASE-NOTES.md:21]:
```suggestion
- Update `critical_service_loop` to throw a runtime error on failure - [#9267] https://github.com/PrefectHQ/prefect/pull/9267
```

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,5 +1,39 @@
...
+
+With the new [Deploy a Prefect flow](https://github.com/marketplace/actions/deploy-a-prefect-flow) Github action, you can automate the build process for deployments orchestrated by Prefect Cloud. The action leverages the new [Projects](https://docs.prefect.io/latest/concepts/projects/) system. See the [action page](https://github.com/marketplace/actions/deploy-a-prefect-flow) for examples and configuration options.
+
+### Cloud Provider Workers
+
+Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are releasing typed workers for each major cloud provider: AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively. 
+
+### Enhancements
+- Add `idempotency_key` to flow runs filter — [#8600](https://github.com/PrefectHQ/prefect/pull/8600)
+- Add details tab to flow run page and increase flow run graph width - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
```
Comment (serinamarie) [inline RELEASE-NOTES.md:15]:
```suggestion
- Add `details` tab to flow run page and increase flow run graph width - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
```

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -1,5 +1,39 @@
 # Prefect Release Notes
 
+## Release 2.10.5
+
+### Deploy a Prefect flow via Github Actions
+
+With the new [Deploy a Prefect flow](https://github.com/marketplace/actions/deploy-a-prefect-flow) Github action, you can automate the build process for deployments orchestrated by Prefect Cloud. The action leverages the new [Projects](https://docs.prefect.io/latest/concepts/projects/) system. See the [action page](https://github.com/marketplace/actions/deploy-a-prefect-flow) for examples and configuration options.
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]:
```suggestion
With the new [Deploy a Prefect flow](https://github.com/marketplace/actions/deploy-a-prefect-flow) Github Action, you can automate the build process for deployments orchestrated by Prefect Cloud. The action leverages the new [Projects](https://docs.prefect.io/latest/concepts/projects/) system. See the [action page](https://github.com/marketplace/actions/deploy-a-prefect-flow) for examples and configuration options.
```
Not totally sure if a singular GitHub Action should be capitalized 🤷‍♀️

Comment (billpalombi) [reply RELEASE-NOTES.md:7]: Good catch. GitHub should be camel case too.

File: RELEASE-NOTES.md [hunk 13]
```diff
@@ -1,5 +1,39 @@
...
 
+## Release 2.10.5
+
+### Deploy a Prefect flow via Github Actions
+
+With the new [Deploy a Prefect flow](https://github.com/marketplace/actions/deploy-a-prefect-flow) GitHub Action, you can automate the build process for deployments orchestrated by Prefect Cloud. The action leverages the new [Projects](https://docs.prefect.io/latest/concepts/projects/) system. See the [action page](https://github.com/marketplace/actions/deploy-a-prefect-flow) for examples and configuration options.
+
+### Cloud Provider Workers
+
+Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are releasing typed workers for each major cloud provider: AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively.
```
Comment (serinamarie) [inline RELEASE-NOTES.md:11]:
```suggestion
Workers, Prefect's next-generation agents, have dedicated infrastructure types. This week, we are releasing typed workers for each major cloud provider: AWS, GCP, and Azure. You will be able to find them in the [prefect-aws](https://github.com/PrefectHQ/prefect-aws), [prefect-gcp](https://prefecthq.github.io/prefect-gcp/), and [prefect-azure](https://github.com/PrefectHQ/prefect-azure) collections, respectively. 

See the following pull requests for implementation details:
- https://github.com/PrefectHQ/prefect-aws/pull/238
- https://github.com/PrefectHQ/prefect-aws/pull/244
- https://github.com/PrefectHQ/prefect-gcp/pull/172
- https://github.com/PrefectHQ/prefect-azure/pull/87
```

File: RELEASE-NOTES.md [hunk 14]
```diff
@@ -1,5 +1,39 @@
...
+- Add `idempotency_key` to flow runs filter — [#8600](https://github.com/PrefectHQ/prefect/pull/8600)
+- Add `details` tab to flow run page and increase flow run graph width - [#9258](https://github.com/PrefectHQ/prefect/pull/9258)
+- Add status code to base client log on retry - [#9265] (https://github.com/PrefectHQ/prefect/pull/9265)
+
+### Fixes
+- Fix issue in which work queues were duplicated in the `default-agent-pool` when creating a deployment - [#9046](https://github.com/PrefectHQ/prefect/pull/9046)
+- Add `configuration` to `Worker.kill_infrastructure` signature - [#9250](https://github.com/PrefectHQ/prefect/pull/9250)
+- Update `critical_service_loop` to throw a runtime error on failure - [#9267] https://github.com/PrefectHQ/prefect/pull/9267
+- Fix pip requirement inference compatibility with Python 3.11+ and pip 23.1+ - [#9278] (https://github.com/PrefectHQ/prefect/pull/9278)
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:23]:
```suggestion
- Fix validation error occurring on default values in `variables` schema of `Workpool.base_job_template` [#9282] (https://github.com/PrefectHQ/prefect/pull/9282)
```

**Review by serinamarie — APPROVED** [review]

**Pull request merged**